### PR TITLE
Extend Trec to DPR Run Converter to use Custom Topics 

### PR DESCRIPTION
**Pull request by ToluClassics**
PR based on https://github.com/castorini/pyserini/issues/1290

### CODE REVIEW

**Comment by lintool** [comment]
Instead of overloading `--topics`, what if we implement _two_ options, `--topics-file` and `--topics-reader` to make more general?

@MXueguang what do you think?

Also, simple test case please.

**Comment by ToluClassics** [comment]
I'd also like to extend this to load custom DPR format kind of QA topics with `DprNqTopicReader` but I'm not sure how to restructure the if-else conditions 🤔. any ideas?

https://github.com/castorini/pyserini/blob/3d8c4731507c3f30e6a88243e26443c681a5c826/pyserini/query_iterator.py#L76-L105

**Comment by MXueguang** [comment]
> Instead of overloading `--topics`, what if we implement _two_ options, `--topics-file` and `--topics-reader` to make more general?

yeah, `--topics-file` and `--topics-reader` looks good.

File: integrations/dense/test_dpr.py [hunk 1]
```diff
@@ -301,9 +303,38 @@ def test_dpr_squad_test_encoded_queries(self):
...
+        self.assertEqual(topic_data["0"]["answers"], ['text'])
+        self.assertEqual(topic_data["0"]["question"], "what is in document three")
+        self.assertEqual(topic_data["1"]["answers"], ['contents'])
+        self.assertEqual(topic_data["1"]["question"], "what is document two")
+
+
     def tearDown(self):
         clean_files(self.temp_files)
 
-
```
Comment (lintool) [inline integrations/dense/test_dpr.py:307]: I think PEP8 wants two empty lines?

**Review by lintool — COMMENTED** [review]
Minor comment from me - but this is @MXueguang 's code I think? so I'll let him take a look also?

**Review by MXueguang — APPROVED** [review]
changes looks good to me

**Pull request merged**